src/keymanager.h

```diff
@@ -134,33 +144,107 @@ lf_keymanager_check_drkey_validity(struct lf_keymanager_key_container *drkey,
 }
 
 /**
- * Derive host-host DRKey from AS-AS DRKey.
+ * Derive HOST-AS DRKey from AS-AS DRKey.
  *
  * @param kmw Keymanager worker context.
  * @param drkey_asas AS-AS DRKey.
  * @param fast_side_host Fast side host address.
- * @param slow_side_host Slow side host address.
- * @param drkey_hh Returning host-host DRKey.
+ * @param drkey_protocol (network byte order).
+ * @param drkey_ha Returning HOST-AS DRKey.
  */
 static inline void
-lf_keymanager_drkey_from_asas(struct lf_keymanager_worker *kmw,
+lf_keymanager_drkey_derive_host_as(struct lf_keymanager_worker *kmw,
 		const struct lf_crypto_drkey *drkey_asas,
 		const struct lf_host_addr *fast_side_host,
+		const uint16_t drkey_protocol, struct lf_crypto_drkey *drkey_ha)
+{
+	assert(LF_HOST_ADDR_LENGTH(fast_side_host) <= LF_CRYPTO_CBC_BLOCK_SIZE);
+
+	uint8_t addr_type_len = (uint8_t)(fast_side_host->type_length);
+	uint8_t *addr_ptr = (uint8_t *)(fast_side_host->addr);
+	uint8_t addr_len = LF_HOST_ADDR_LENGTH(fast_side_host);
+
+	// IPv4 mapped to IPv6
+	if (addr_type_len == 3 && *(uint64_t *)(fast_side_host->addr) == 0 &&
+			*(uint32_t *)(fast_side_host->addr + 8) == 0xffff0000) {
```

Comment:
Use ippercase for hex consts like below?

Reply:
Done